src/include/btree.h

```diff
@@ -93,6 +93,9 @@ typedef enum { /* Start position for eviction walk */
 /* An invalid btree file ID value. ID 0 is reserved for the metadata file. */
 #define WT_BTREE_ID_INVALID UINT32_MAX
 
+/* Three bits reserved for ID namespaces. */
+#define WT_BTREE_ID_NAMESPACE_SHARED ((uint32_t)1 << 29)
```

Comment:
Does this mean if we run wt locally, we also cannot use these ids?

Reply:
I guess if that's the case, it is also not backward compatible.

Reply:
The PR in its current form is technically backwards-incompatible, due to the assertion added to the existing code for importing tables. In practice I don't think you'd hit this given you'd have to be importing from a database that had over one billion tables.

But more generally: yes, I think this is intended as an incompatible change. It was a design choice that the namespace eats into the previously-permitted ID range for local tables.